Jaden Taking Over For Peterson: Implement No Teams Found Message - Teams Page (DONE Jaden)

peterson337 opened the pull request:
Jaden Taking Over For Peterson: Implement No Teams Found Message - Teams Page (DONE Jaden) #3928

# Description
This open PR was created to implement a message when no team is found in the teams table.

## Related PRS (if any):
None

## Main changes explained:
The CreateNewTeamPopup.jsx component has been modified to fix the bug.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. Log in as an admin or owner user.
5. Go to Other Links → Teams
6. Click on the input above the table, type the name of a team that doesn't exist, and a message with a button should appear. After clicking the button, a modal will open and the input should receive the name of the non-existent team.

## Screenshots or videos of changes:
<img width="1059" height="258" alt="image" src="https://github.com/user-attachments/assets/c8b8a952-1b38-46ce-91f9-0066f17cdf51" />
<img width="938" height="364" alt="image" src="https://github.com/user-attachments/assets/dd6f274c-303e-41ed-9a1e-d3c2e1cb778f" />
<img width="1008" height="353" alt="image" src="https://github.com/user-attachments/assets/cf31176c-b2e0-4dbc-a771-28180ca3ee2f" />
https://github.com/user-attachments/assets/b27b8f8f-7111-4bc1-9dca-0e151720e584

ChiragBellara commented:
Reviewed PR #3928 

1. Message box to create a team that does not exist shows up as expected.
<img width="720" height="450" alt="message box for team not found appears" src="https://github.com/user-attachments/assets/9d512838-ab15-4ca3-8098-8835a02b2bcb" />

2. On clicking "Create Team", modal opens up with the team name. On save, New Team gets created and saved successfully.
<img width="720" height="450" alt="modal with team name opens up" src="https://github.com/user-attachments/assets/ab6c0a64-a4d0-4041-a78e-c385c69ba44f" />
<img width="720" height="450" alt="team gets created and saved" src="https://github.com/user-attachments/assets/5cae9a12-886b-4716-8f5d-497e20ab0b04" />

3. Changes work in both light and dark mode similarly.
<img width="720" height="450" alt="changes in dark mode" src="https://github.com/user-attachments/assets/a3515623-dfa1-4bf9-8d99-c50ed990730f" />

Jaden300 committed:
Add no-teams-found message with Create Team button on Teams page

When a search yields no results, show a message and a button that
opens the Create New Team modal with the search text pre-filled.

Jaden300 committed:
Use setState callback when referencing previous state in onCreateNewTeamFromSearch

DeepighaJ reviewed (approved):
- Tested the PR locally. The functionality works as expected.
- Verified a non existing team displays message with create Team button.
- Verified on clicking the create team button a model opens with same team name populated automatically.
- Verified successful team creation with success toast.

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/be4b4ab1-df9a-447b-ba0e-07c9dbc2bdea" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/41790546-02fb-4baf-a0cb-614ad10597fd" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/1ffc8bcd-162b-498b-b9e4-52763ba694d2" />
